test/Microsoft.AspNetCore.Mvc.Razor.Test/ViewComponentTagHelpers/ViewComponentTagHelperDescriptorFactoryTest.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Razor.ViewComponentTagHelpers;
+using Microsoft.AspNetCore.Mvc.ViewComponents;
+using Microsoft.AspNetCore.Razor.Compilation.TagHelpers;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Test.ViewComponentTagHelpers
+{
+    public class ViewComponentTagHelperDescriptorFactoryTest
+    {
+        public static TheoryData<string, IEnumerable<TagHelperDescriptor>> AssemblyData
+        {
+            get
+            {
+                var provider = new TestViewComponentDescriptorProvider();
+
+                var assemblyOne = "Microsoft.AspNetCore.Mvc.Razor";
+                var assemblyTwo = "Microsoft.AspNetCore.Mvc.Razor.Test";
+                var assemblyNone = "";
+
+                return new TheoryData<string, IEnumerable<TagHelperDescriptor>>
+                {
+                    { assemblyOne, new List<TagHelperDescriptor> { provider.tagHelperDescriptorOne } },
+                    { assemblyTwo, new List<TagHelperDescriptor> { provider.tagHelperDescriptorTwo } },
+                    { assemblyNone, new List<TagHelperDescriptor>() }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(AssemblyData))]
+        public void CreateDescriptors_ReturnsCorrectDescriptors_ForViewComponentsInAssembly(
+            string assemblyName, 
+            IEnumerable<TagHelperDescriptor> expectedDescriptors)
+        {
+            // Arrange
+            var provider = new TestViewComponentDescriptorProvider();
+            var factory = new ViewComponentTagHelperDescriptorFactory(provider);
+
+            // Act
+            var descriptors = factory.CreateDescriptors(assemblyName);
+
+            // Assert
+            Assert.Equal(descriptors.Count(), expectedDescriptors.Count());
+            
+            if (descriptors.Count() > 0) // Either empty or 1 descriptor.
```

Comment:
Instead of doing this bit, I'd just change the data to only have valid assembly names so you result in > 0 descriptors.

You can then add a second test that validates assembly names that result in nothing.